luxonis_train/utils/config.py

```diff
@@ -32,32 +44,32 @@ class MetricModuleConfig(AttachedModuleConfig):
 
 class FreezingConfig(BaseModelExtraForbid):
     active: bool = False
-    unfreeze_after: int | float | None = None
+    unfreeze_after: NonNegativeInt | NonNegativeFloat | None = None
```

Comment:
How are floats actually handled? E.g. if I set unfreeze_after=2.3 will if actually unfreeze after 2.3 epochs or does it round the number? If the latter then we can just set so we only accept int

Reply:
Float specifies the percentage of the num epochs. So with 100 epochs and `unfreeze_after` set to 0.2, it would unfreeze after 20 epochs.